internal/route/routes.go

```diff
@@ -93,40 +95,29 @@ func HandleTodos(w http.ResponseWriter, r *http.Request) {
 
 // HandleTodoDone marks todo as done or undone
 func HandleTodoDone(w http.ResponseWriter, r *http.Request) {
-	if r.Method == "GET" {
+	if r.Method != "PUT" {
 		w.WriteHeader(http.StatusForbidden)
-		log.Fatalf("[error] Route only handles POST requests")
+		log.Fatalf("[error] Route only handles PUT requests")
 		return
 	}
 
-	b, err := ioutil.ReadAll(r.Body)
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		log.Fatalf("[error] failed to read request body: %v\n", err)
-	}
-
-	defer r.Body.Close()
-
-	todo := &model.Todo{}
-	err = json.Unmarshal(b, &todo)
-
-	if err != nil {
-		w.WriteHeader(http.StatusInternalServerError)
-		log.Fatalf("[error] failed to unmarshal json: %v\n", err)
-	}
-
 	vars := mux.Vars(r)
 	listID := vars["listId"]
 	todoID := vars["todoId"]
 
-	todo, err = model.MarkTodoAsDone(listID, todoID, todo.Done)
+	t, err := model.TodoByID(listID, todoID)
+	if err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		log.Fatalf("[error] failed to fetch todo: %v\n", err)
+	}
 
+	err = t.ToggleDone()
 	if err != nil {
 		w.WriteHeader(http.StatusInternalServerError)
-		log.Fatalf("[error] failed to update todo: %v\n", err)
+		log.Fatalf("[error] failed to toggle done for todo: %v\n", err)
 	}
```

Comment:
So here we use the new `TodoByID` to grab the todo by the IDs in the URL then just call `ToggleDone()` on it.